### PR TITLE
Backoff wrapper

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

For https://github.com/istio/istio/issues/40626

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
cc @howardjohn @ramaraochavali

**Comment by hzxuzhonghu** [comment]
I will replace all github.com/cenkalti/backoff/v4 usage if it makes sense

**Comment by ramaraochavali** [comment]
LGTM

File: pkg/backoff/exponential.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+// ExponentialBackOff is a wrapper of backoff.ExponentialBackOff to override its NextBackOff().
+type ExponentialBackOff struct {
+	*backoff.ExponentialBackOff
+}
+
+// NewExponentialBackOff creates an istio wrapped ExponentialBackOff.
+func NewExponentialBackOff(initFuncs ...func(off *ExponentialBackOff)) BackOff {
+	b := &ExponentialBackOff{}
```
Comment (howardjohn) [inline pkg/backoff/exponential.go:42]: Do we need a pointer? The inner struct is already a pointer

**Review by howardjohn — APPROVED** [review]

File: pkg/backoff/exponential.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/backoff/exponential.go:42]: Not a must

Comment (howardjohn) [reply pkg/backoff/exponential.go:42]: IMO wrapper should not be a pointer, can we swap it?

Comment (hzxuzhonghu) [reply pkg/backoff/exponential.go:42]: Done in the following

**Pull request merged**